### ISSUE TITLE
Unable to add filesystem custom storage volume on ceph pool to a VM

### ISSUE THREAD

**Issue by stgraber** [lxc/lxd#11351]
LXD seems to incorrectly be going down the RBD block volume path when attaching a filesystem volume from a ceph pool to a VM.

Reproducer looks like this, assuming two pools, `local` being ZFS and `remote` being Ceph:

 - lxc init --storage local images:ubuntu/22.04/cloud v1 --vm
 - lxc storage volume create remote foo size=10GiB
 - lxc config device add v1 foo disk pool=remote source=foo path=/mnt/foo

On startup, you'll get a confusing:
```
Error: Failed setting up device via monitor: Failed adding block device for disk device "foo": Failed adding block device: error reading header from custom_demo_foo.block: No such file or directory
Try `lxc info --show-log v1` for more info
```

Adding a bit of debug to dump the Mounts being processed, we see:
```
ERROR  [2023-02-10T14:09:30-05:00] stgraber: monhook: {DevName:foo DevPath:rbd lxd_dakara/foo id=admin:pool=lxd_dakara:conf=/etc/ceph/ceph.conf TargetPath: FSType: Opts:[] Freq:0 PassNo:0 OwnerShift:} 
```

Which looks quite incorrect. Instead what we should have is LXD call MountCustomVolume to have the host mount that RBD volume, and then pass it into the VM through 9p/virtio-fs, which should then skip the QMP disk path and instead rely on good old fashion qemu.conf entries.

**Comment by stgraber** [comment]
@monstermunchkin @tomponline not sure which one of you is the most fluent in that part of the code these days.
I poked at it for a few minutes but couldn't immediately find a fix. Not exactly sure how far up the stack the issue is, it may be all the way into the disk device handling?

**Comment by monstermunchkin** [comment]
@stgraber did this work before?

**Comment by stgraber** [comment]
@monstermunchkin no idea.
It's just something I tried to do for the recent video on disk devices and when it blew up, I had to re-record without that segment :)

We do the right thing for other filesystem volumes, if I create one on local storage, it will work just fine when attached to a VM.
I suspect this may have worked fine until we either added the optimized RBD logic to the qemu driver (have QEMU directly perform the RBD access) or maybe we broke it when adding the QMP based logic to setup storage.
